src/IRutils.cpp

```diff
@@ -365,36 +365,36 @@ std::string resultToSourceCode(const decode_results *results) {
   // Only display the value if the decode type doesn't have an A/C state.
   if (!hasACState(results->decode_type))
     output += " " + uint64ToString(results->value, 16);
-  output += "\n";
+  output += F("\n");
 
   // Now dump "known" codes
   if (results->decode_type != UNKNOWN) {
     if (hasACState(results->decode_type)) {
 #if DECODE_AC
       uint16_t nbytes = results->bits / 8;
-      output += "uint8_t state[" + uint64ToString(nbytes) + "] = {";
+      output += String(F("uint8_t state[")) + uint64ToString(nbytes) + String(F("] = {"));
       for (uint16_t i = 0; i < nbytes; i++) {
-        output += "0x";
-        if (results->state[i] < 0x10) output += "0";
+        output += String(F("0x"));
+        if (results->state[i] < 0x10) output += F("0");
         output += uint64ToString(results->state[i], 16);
         if (i < nbytes - 1) output += ", ";
       }
-      output += "};\n";
+      output += F("};\n");
 #endif  // DECODE_AC
     } else {
       // Simple protocols
       // Some protocols have an address &/or command.
       // NOTE: It will ignore the atypical case when a message has been
       // decoded but the address & the command are both 0.
       if (results->address > 0 || results->command > 0) {
-        output += "uint32_t address = 0x" +
-                  uint64ToString(results->address, 16) + ";\n";
-        output += "uint32_t command = 0x" +
-                  uint64ToString(results->command, 16) + ";\n";
+        output += String(F("uint32_t address = 0x")) +
```

Comment:
Again, ditto. Please don't use `String()` unless we have to. (I'll stop flagging these)